lib/simple_forms_api/form_remediation.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  module FormRemediation
+    class Error < StandardError
+      attr_reader :base_error, :details
+
+      DEFAULT_MESSAGE = 'An error occurred during the form remediation process'
+
+      def initialize(message: DEFAULT_MESSAGE, error: nil, details: nil, backtrace: nil)
+        super(message)
+        @base_error = error
+        @details = details
+        @custom_backtrace = backtrace
+      end
+
+      def message
+        [super, custom_message].compact.join(' - ')
+      end
+
+      def backtrace
+        @custom_backtrace || base_error&.backtrace || super
+      end
+
+      private
+
+      def custom_message
+        details.is_a?(Hash) ? details[:message] : base_error&.message
```

Comment:
The reason we need this class is for this `custom_message`, right? I like it!

Reply:
That's the main reason! Also, it gives us modular control of handling when just this service fails.